### ISSUE TITLE
Combined Product Pipelines not working with external versions/products

### ISSUE THREAD

**Issue by robbrad** [awslabs/aws-service-catalog-factory#263]
When the portfolio is defined as 

```
Schema: factory-2019-04-01
Portfolios:
  - DisplayName: central-it-team-portfolio
    Description: A place for self service products ready for your account
    ProviderName: central-it-team
    Associations:
      - arn:aws:iam::${AWS::AccountId}:role/Admin
    Products:
      - Name: account-vending-account-creation
        Owner: central-it@customer.com
        Description: template used to interact with custom resources in the shared projects
        Distributor: central-it-team
        SupportDescription: Contact us on Chime for help #central-it-team
        PipelineMode: combined
        SupportEmail: test@test.com
        SupportUrl: http://test
```

Products is a child element of Portfolios and this never enters the generate products loop defined here(skipping the combined option):

https://github.com/awslabs/aws-service-catalog-factory/blob/master/servicecatalog_factory/commands/portfolios.py#L273

With the following portfolio schema - "version" pipelines are created but combined are ignored

[ignored.zip](https://github.com/awslabs/aws-service-catalog-factory/files/8718478/ignored.zip)

If we change the code here 
https://github.com/awslabs/aws-service-catalog-factory/blob/f97030f710e908189e0eb650f0303be73c5e3b04/servicecatalog_factory/commands/portfolios.py#L273

To

`for product in portfolios['Portfolios'][0].get("Products", []):`

(Note: will break the schema where products and portfolios are at the same level)

**Version pipelines get created** and then **combined are also created** 

However the **CreateCombinedProductPipelineTemplateTask** class fails with TypeError: 'NoneType' object is not iterable

![image](https://user-images.githubusercontent.com/6122503/169103269-bfbf463b-ef67-43a8-9d9f-8f055551213c.png)

![image](https://user-images.githubusercontent.com/6122503/169102286-d4be39fd-e3a2-41cc-b05c-74babf0d66ba.png)

Results
[results.zip](https://github.com/awslabs/aws-service-catalog-factory/files/8718565/results.zip)

**Comment by eamonnfaherty** [comment]
Thanks for the details when raising this.  I hope to look at this next week as part of a new feature I am building out.

**Comment by robbrad** [comment]
Great - you should be able to recreate using the "ignored.zip"

**Comment by eamonnfaherty** [comment]
released in 0.82.0
